### ISSUE TITLE
Java14 Full Records Support Check Validation: CommentsIndentationCheck

### ISSUE THREAD

**Issue by nmancus1** [checkstyle/checkstyle#8489]
Child of #8452 
Check documentation: https://checkstyle.sourceforge.io/config_misc.html#CommentsIndentation

From check documentation:
_Controls the indentation between comments and surrounding code. Comments are indented at the same level as the surrounding code. Detailed info about such convention can be found here_

```
➜  full-record-grammar /usr/lib/jvm/java-14-openjdk/bin/javac --enable-preview --source 14 TestClass.java
Note: TestClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.
➜  full-record-grammar cat config.xml    
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="CommentsIndentation"/>
  </module>
</module>
➜  full-record-grammar cat TestClass.java
class TestClass {
    public record MyRecord(int x) {
        public void myMethod(){
            String breaks = "J"
            // violation (comment should have the same indentation level as line 6)
                + "A"
                // it is OK
                + "V"
                + "A"
            // it is OK
            ;
        }
    }

    public record MyOtherRecord() {
   // comment
   //  block
   // violation (comment should have the same indentation level as line 19)
    }
            ///////////// violation, should be same as line 21
    public record myOtherOtherRecord () {

    }
}


➜  full-record-grammar java $RUN_LOCALE -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.35-SNAPSHOT-all.jar -c config.xml TestClass.java
Starting audit...
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:5:13: Comment has incorrect indentation level 12, expected is 16, indentation should be the same level as line 6. [CommentsIndentation]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:18:4: Comment has incorrect indentation level 3, expected is 4, indentation should be the same level as line 19. [CommentsIndentation]
[ERROR] /home/nick/Desktop/full-record-grammar/TestClass.java:20:13: Comment has incorrect indentation level 12, expected is 4, indentation should be the same level as line 21. [CommentsIndentation]
Audit done.
Checkstyle ends with 3 errors.


```

This check works as intended, so I think we can close this one.

**Comment by pbludov** [comment]
One corner case is missing:
```java
class TestClass {
    class
       Wrapped { }
       // violation
}
    

class TestClass2 {
    record
        Wrapped() { }
        // expected violation
}
```
produces
```
Starting audit...
[ERROR] /home/pbludov/src/tmp/TestClass.java:4:8: Comment has incorrect indentation level 7, expected is 4, indentation should be the same level as line 2. [CommentsIndentation]
Audit done.
```

The method `CommentsIndentationCheck.isDefinition` should be aware of `TokenTypes.RECORD_DEF`.
